helper/src/main/java/me/lucko/helper/utils/XMaterial.java

```diff
@@ -0,0 +1,1686 @@
+        if (!version.startsWith("VERSION_")) version = "VERSION_" + version;
+        String check = version;
+        return Arrays.stream(MinecraftVersion.VALUES).anyMatch(v -> v.name().equals(check)) ? MinecraftVersion.valueOf(version) : MinecraftVersion.UNKNOWN;
+    }
+
+    /**
+     * Checks if the material can be damaged from {@link #DAMAGEABLE}.
+     *
+     * @param name the name of the material.
+     * @return true of the material can be damaged.
+     */
+    public static boolean isDamageable(@Nonnull String name) {
+        Validate.notEmpty(name, "Material name cannot be null or empty");
+        return Arrays.stream(DAMAGEABLE).anyMatch(name::contains);
+    }
+
+    public static boolean isDuplicated(@Nonnull String name) {
+        Validate.notEmpty(name, "Material name cannot be null or empty");
+        String formatted = format(name);
+        return DUPLICATED.entrySet().stream().anyMatch(x -> x.getKey().name().equals(formatted));
+    }
+
+    /**
+     * Converts the material's name to a string with the first letter uppercase.
+     *
+     * @return a converted string.
+     */
+    @Nonnull
+    public String toWord() {
+        return toWord(this.parseMaterial().name());
+    }
+
+    /**
+     * Gets the ID (Magic value) of the material.
+     * If an {@link IllegalArgumentException} was thrown from this method,
+     * you should definitely report it.
+     *
+     * @return the ID of the material. -1 if it's a new block.
+     */
+    @SuppressWarnings("deprecation")
+    public int getId() {
+        return this.isNew() ? -1 : this.parseMaterial().getId();
+    }
+
+    public boolean isDuplicated() {
+        return DUPLICATED.containsKey(this);
+    }
+
+    /**
+     * Checks if the given string matches any of this material's legacy material names.
+     *
+     * @param name the name to check
+     * @return true if it's one of the legacy names.
+     */
+    public boolean matchAnyLegacy(String name) {
+        String formatted = format(name);
+        return Arrays.asList(this.legacy).contains(formatted);
+    }
+
+    /**
+     * Checks if the item is duplicated for a different purpose in new versions from {@link #DUPLICATED}.
+     *
+     * @return true if the item's name is duplicated.
+     * @see #matchXMaterial(String, byte)
+     */
+    @Nullable
+    public XMaterial getXMaterialIfDuplicated() {
+        Optional<Map.Entry<XMaterial, XMaterial>> mat = DUPLICATED.entrySet().stream().filter(m -> m.getKey() == this).findFirst();
+        return mat.map(Map.Entry::getValue).orElse(null);
+    }
+
+    /**
+     * @return true if the item can be damaged.
+     * @see #isDamageable(String)
+     */
+    public boolean isDamageable() {
+        return isDamageable(this.name());
+    }
+
+    /**
+     * Get the {@link ItemStack} data of this material in older versions.
+     * Which can be accessed with {@link ItemStack#getData()} then MaterialData#getData()
```

Comment:
Additional paragraphs in comments should include an empty line between previous paragraphs.